### PR TITLE
[#158068259] fetch transactions and cards from api

### PR DESCRIPTION
**Pull request by fgiobergia**
This PR introduces the PagoPA mocked API (currently available @ https://pagopa-mock-rest.eu-gb.mybluemix.net/ ). These are based on the swagger file made available by PagoPA (not online any longer -- a copy can be seen [here](https://pagopa-mock-rest.eu-gb.mybluemix.net/api-docs)). 

This PR makes use of `GET /wallet` and `GET /transactions` to fetch a list of available payment methods (only credit cards supported atm) & transactions available. 

The fields returned by the API have been interpreted at the best of my capabilities. The only open point regards the "merchant" field, which is currently being considered as the "recipient" of the payment (no other field has been found that can be interpreted as the recipient).

### CODE REVIEW

**Comment by fgiobergia** [comment]
About the failing checks: `gen-api-models` is converting arrays to `t.array`, while it should actually be `t.array(t.string)`. This is making the checks fail. I didn't have the time to investigate about what's going on with gen-api-models yet.

**Comment by cloudify** [comment]
@gunzip see https://github.com/teamdigitale/italia-app/pull/264#issuecomment-401436924

File: ts/api/pagopa.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  const pCard: PagoPACreditCard = pWallet.creditCard;
+  if (
+    pWallet.lastUsage === undefined ||
+    pWallet.idWallet === undefined ||
+    pCard.pan === undefined ||
+    pCard.holder === undefined ||
+    pCard.expireMonth === undefined ||
+    pCard.expireYear === undefined ||
+    pCard.brandLogo === undefined ||
+    CreditCardType.decode(pCard.brandLogo).isLeft()
```
Comment (gunzip) [inline ts/api/pagopa.ts]: you can use `!CreditCardType.is(pCard.brandLogo)`

File: ts/api/pagopa.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      pCard.expireYear
+    }` as NonEmptyString,
+    brandLogo: CreditCardType.decode(pCard.brandLogo).value as CreditCardType // tslint:disable-line no-useless-cast
+  };
+};
+
+const mapToLocalTransaction = (
+  pTransaction: PagoPATransaction
+): WalletTransaction => {
+  const amount = Amount.decode(pTransaction.amount).getOrElse({
```
Comment (gunzip) [inline ts/api/pagopa.ts]: you can use `Amount.decode(pTransaction.amount).toUndefined()`

File: ts/api/pagopa.ts (hunk 1 above)
Comment (gunzip) [reply ts/api/pagopa.ts]: btw it looks like all these properties are mandatory in the CreditCard type below, so I guess `CreditCard.is(pCard)` should suffice (or `CreditCard.decode(pCard).isLeft()`)

**Comment by gunzip** [comment]
@fgiobergia @cloudify about

`About the failing checks: gen-api-models is converting arrays to t.array, while it should actually be t.array(t.string). This is making the checks fail. I didn't have the time to investigate about what's going on with gen-api-models yet.`

it's not clear to me how to reproduce the error since the endpoint feeded to gen-api-specs here is http://pagopa-mock-rest.eu-gb.mybluemix.net/api-docs which is a json while gen-api-specs expects a yaml. may you tell me the exact command line ?

**Comment by fgiobergia** [comment]
@gunzip actually, gen-api-specs apparently accepts the json as well and correctly generates most of the types, eccept for cases like:
```javascript
    "LogicDelete": {
      "type": "object",
      "properties": {
        "email": {
          "type": "string"
        },
        "userStatusEnum": {
          "type": "array",
          "items": {
            "type": "string",
            "enum": [
              "ANONYMOUS",
              "REGISTERED",
              "REGISTERED_SPID",
              "NEED_OTP",
              "NEED_PASSWORD",
              "WAITING_NEW_PASSWORD",
              "DELETED"
            ]
          }
        }
      },
```
Where it generates (for `userStatusEnum`) a `t.array` and does not check `items.type` (and thus does not define a type for the array -- thus causing an error)

**Comment by fgiobergia** [comment]
Superseded by #339

**Pull request closed without merging**